frontends/p4/typeChecking/typeChecker.cpp

```diff
@@ -2768,8 +2816,10 @@ void TypeInference::checkCorelibMethods(const ExternMethod* em) const {
 
             auto arg0 = mce->arguments->at(0);
             auto argType = typeMap->getType(arg0, true);
-            if (!argType->is<IR::Type_Header>() && !argType->is<IR::Type_Dontcare>()) {
-                typeError("%1%: argument must be a header", mce->arguments->at(0));
+            if (!argType->is<IR::Type_Header>() &&
+                !argType->is<IR::Type_Dontcare>()) {
+                typeError("%1%: argument must be a header/bit-vector struct",
+                          mce->arguments->at(0));
```

Comment:
The error message says bit-vector structs are allowed, but that is not being checked in the  if-condition. I'm not convinced that anything here needs to change at all, as I don't see a reason to extract into anything but header.

Reply:
This error message is leftover from when the method had changed to support a struct with extract().  After comments for extract() I backed out the changes but missed this one.   In the above code changes, the code logic hasn't changed.  Just the string "/bit-vector struct" needs to be removed from the error.